### PR TITLE
Add 2022q4 Azure report.

### PR DESCRIPTION
**Pull request by lwhsu**

### CODE REVIEW

File: 2022q4/azure.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+=== FreeBSD on Microsoft HyperV and Azure
+
+Links: +
+link:https://wiki.freebsd.org/MicrosoftAzure[Microsoft Azure article on FreeBSD wiki] URL: link:https://wiki.freebsd.org/MicrosoftAzure[] +
+link:https://wiki.freebsd.org/HyperV[Microsoft HyperV article on FreeBSD wiki] URL: link:https://wiki.freebsd.org/HyperV[]
```
Comment (pauamma) [inline 2022q4/azure.adoc:5]: `URL: link:https://wiki.freebsd.org/MicrosoftAzure[https://wiki.freebsd.org/MicrosoftAzure]` and likewise for the next one.

File: 2022q4/azure.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Contact: Wei Hu <whu@FreeBSD.org> +
+Contact: Li-Wen Hsu <lwhsu@FreeBSD.org> +
+
+In this quarter, the link:https://azuremarketplace.microsoft.com/marketplace/apps/thefreebsdfoundation.freebsd-12_4[12.4-RELEASE image] has been published to Azure Marketplace.
+
+Work in progress tasks:
+
+* Automating the image building and publishing process and merge to src/release/.
+* Building and publishing ZFS-based images to Azure Marketplace
+** All the requirted codes are merged to main branch, and can create ZFS-bared image by specifying `VMFS=zfs`.
```
Comment (pauamma) [inline 2022q4/azure.adoc:19]: s/requirted/required/

What does "ZFS-bared" mean?

s/image/images/, I think.

File: 2022q4/azure.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+** Need to make the build process more automaitc and collaborating with release engineering to start generating snapshots.
```
Comment (pauamma) [inline 2022q4/azure.adoc:20]: s/automaitc/automatic/

File: 2022q4/azure.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+* Fixing booting issue on Hyper-V gen2 VM in Azure
+** https://bugs.freebsd.org/264267
+* Porting Hyper-V guest support to aarch64
+** https://bugs.freebsd.org/266248
+** https://bugs.freebsd.org/267654
+
+Open tasks:
+
+* Update FreeBSD related doc at link:https://docs.microsoft.com[]
```
Comment (pauamma) [inline 2022q4/azure.adoc:36]: `link:https://docs.microsoft.com[https://docs.microsoft.com]` (or more explicit link if there's a specific subdir for FreeBSD)

**Review by pauamma — CHANGES_REQUESTED** [review]

**Comment by lwhsu** [comment]
I think repeat the same URL in the square brackets isn't very useful, and may even cause some editing error in the future.  Just using the `link:` macro and `[]` to explicitly mark the URL part should be enough for all the cases, including the special characters.

Also, I just found that this seems what other files do.  I am OK to add to this one, but I strongly suggest let's remove to a simpler syntax, in this, or at least the next quarter.

**Comment by grahamperrin** [comment]
> … repeat the same URL in the square brackets …

If I understand correctly, the repetition is currently to _add links to the mail version_: 

* https://github.com/freebsd/freebsd-quarterly/issues/530#issuecomment-1278250578

**Comment by grahamperrin** [comment]
From <https://github.com/freebsd/freebsd-quarterly#readme>, with emphasis: 

> … please, **do not squash or force-push** …

**Review by pauamma — APPROVED** [review]
NTS: fill in empty []s.

**Comment by lwhsu** [comment]
> From https://github.com/freebsd/freebsd-quarterly#readme, with emphasis:
> 
> > … please, **do not squash or force-push** …

Apologies, didn't notice this policy change. BTW, force-push on feature branch is quite common and useful to keep the history (of the main branch) clean, this is widely used in many other pleases.  Also the review feature of GitHub is improved to cope with it.  Please seriously consider this.

cc @lsalvadore

**Comment by lsalvadore** [comment]
The reviewing procces is mainly in @pauamma's and @grahamperrin's hands, so I think they should choose what works best for them about squashing and force-pushing. Personally, I have no preference.

A few thoughts about the topic:

- starting from the next quarter, if everything  goes as planned, authors will be able to submit their reports on Phabricator too. Things such as squashing and force-pushing are not part of the usual Phabricator workflow: authors will submit their work, modify their files following the reviewing process and only one single commit to the docs repository will end the review. Further changes will require more commits and could not be squashed in the past commits;

- if one wants a cleaner history in the main project, it is not necessary to force-push anything in pull requests: it is enough to select the "Squash and merge" option in GitHub when merging the pull request and only one commit will result. And using git from command line allows for even more flexibility (and then GitHub might not automatically recognize the pull request merging: this often happens to me. Not a big deal). However I also understand that some report authors might prefer to keep a clean history with multiple commits, which would require to force-push.

- policies are useful and important, but they also need to be applied with care for special cases. See for example my recent pull request #562: I only had to fix a typo in a commit message, without modifying any file. It would have been absurd to merge the pull request including the commit message with the typo instead of amending it and force-pushing;

- attempting to keep a clean history is nice, but is also difficult. Indeed, what happens in pull requests is only the first main reviewing process, but later the reports are reviewed again. I review them once again when I put them together and render them, correcting a few things here and there as needed (mostly adoc syntax and typos). Then they can be reviewed again when I open the Phabricator review to commit the changes to the docs repository. And the final review is done by our readers, who are often able to spot a few more things after publication in spite of our best efforts. Thus, many reports happen to get more than one commit even if they were created with only one clean single commit pull request;

- I think we all agree that a cleaner history is better. However, we should not forget that if we use a VCS it is also to keep track of who does what and why. And also to change our mind if we want to. Cleaning history risks to go against these goals.

**Comment by grahamperrin** [comment]
Thanks, all

> … choose what works best for them about squashing and force-pushing. 

I'm aware of the plea in https://github.com/freebsd/freebsd-quarterly/blob/main/README.adoc only because I helped to edit the file. 

> Personally, I have no preference. …

I don't know enough about Git to express a preference. 

If someone can point me to a case where a squash or force-push _did_ cause difficulty, I might form a preference …

**Pull request merged**